study1/week1/unique-binary-search-trees.java

```diff
@@ -0,0 +1,22 @@
+// https://leetcode.com/problems/unique-binary-search-trees/
+// References: https://st-lab.tistory.com/300
+class Solution {
+    // Time Complexity: O(n^2)
+    // Space Complexity: O(n^2)
```

Comment:
The space complexity for this function is `O(n)`, each call contains only constant number of variables, and the recursion depth will only be at maximum n at any point of time, because in the worst case it'll only call numTrees(n)->numTrees(n-1) -> .... thus the maximum call stack will contain `n` calls.